src/main/java/com/epam/book/dao/specification/AuthorSpecification.java

```diff
@@ -0,0 +1,23 @@
+package com.epam.book.dao.specification;
+
+import com.epam.book.dao.DaoException;
+import com.epam.book.model.Book;
+
+public class AuthorSpecification<T> extends AbstractSpecification<T> {
+
+    @Override
+    public boolean matchesField(Book book, T value) throws DaoException {
+
+        String author = book.getAuthor();
+        String comparableValue;
+
+        if (value instanceof String) {
```

Comment:
AuthorSpecification<T> extends AbstractSpecification<String>, in this case you won't need  instance of check and throwing exception either